book/src/article/generate_multiple_files.md

```diff
@@ -183,4 +183,74 @@ gen:
         --rust-output generated_api_apple generated_api_orange
 ```
 
-NOTE: the suffix `apple` and `orange` are quite disordered for each flag here on purpose. It is compilable and usable. But as you should know, it is not a good practice, semantically. It is all up to you to decide the field names for each flag, so be beware of it!
\ No newline at end of file
+NOTE: the suffix `apple` and `orange` are quite disordered for each flag here on purpose. It is compilable and usable. But as you should know, it is not a good practice, semantically. It is all up to you to decide the field names for each flag, so beware of it!
+
+## More about the auto-generated shared block
+
+### Shared Rust files
+
+As mentioned above, flutter_rust_bridge would automatically generate a new file called `bridge_generated_shares.rs` for the shared types --- To make it clear and short, let's call all the related generated stuff(the Rust files, Dart files, and C files) the shared (Api) block; and on the contrary, those normally generated stuff is called regular (Api) block.
+
+In a shared block, there is stuff including any shared types like primitive, customized defined types(structs, enums)list(vector), even the type is wrapped with `option`, or `SyncReturn`. And note: For a shared customized type, please DON'T define it in any regular Api block.
```

Comment:
nit: is it possible we automatically check and throw error about this?
(even if yes, we may implement it in the future, not in this PR!)

Reply:
I did think of it before. And technically, it should be possible to check it. So I agree and leave it to future pr. [711ee3f](https://github.com/fzyzcjy/flutter_rust_bridge/pull/1145/commits/711ee3f6e534a8b29f0e74b875de1546c39619eb)